modules/ting_search_ortograf/includes/ting_search_ortograf.ajax.inc

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * @file
+ * AJAX functions bibliotek.dk autocomplete.
+ */
+
+/**
+ * Auto-complete callback for the ting search block.
+ *
+ * The function returns JSON and ending execution.
+ *
+ * @param string $string
+ *   The string to search suggestion with.
+ */
+function ting_search_ortograf_callback($string = '') {
+  $string = check_plain(strtolower($string));
+  $query = ting_search_ortograf_settings();
+  $minimum_string = (int) $query['minimumString'];
+  if (drupal_strlen($string) < $minimum_string) {
+    drupal_json_output(array());
+    drupal_exit();
+  }
+  // MinimumString is not a suggestion parameter.
+  unset($query['minimumString']);
+  $query['q'] = $string;
+
+  // Try to use cache for the search string.
+  $cid = md5(serialize($query));
+  if ($cache = cache_get($cid, 'cache_ting_search_ortograf')) {
+    $matches = $cache->data;
+  }
+  else {
+    $matches = ting_search_ortograf_get_suggestions($query);
+    // Store the matches in cache for faster lookup next time.
+    cache_set($cid, $matches, 'cache_ting_search_ortograf', CACHE_TEMPORARY);
```

Comment:
Is `CACHE_TEMPORARY` really the right approach here? It seems as if the index we search in is quite stable as it works across all libraries and thus not very prone to changes. A longer cache lifetime would lead to quicker response times which I would consider important for autocompletion

Reply:
Autocomplete responses are not entirely stable - new titles are added on a daily basis. 
Cron jobs (and cache clear) are set to run once daily, which sounds about right to me.

Reply:
:+1: